gateway/src/shard/processor/impl.rs

```diff
@@ -199,20 +232,20 @@ impl ShardProcessor {
                 #[cfg(feature = "metrics")]
                 counter!("GatewayEvent", 1, "GatewayEvent" => "InvalidateSessionFalse");
                 debug!("[EVENT] InvalidateSession(false)");
-                self.reconnect().await;
+                self.reconnect(false).await;
             },
             Reconnect => {
                 #[cfg(feature = "metrics")]
                 counter!("GatewayEvent", 1, "GatewayEvent" => "Reconnect");
                 debug!("[EVENT] Reconnect");
-                self.reconnect().await;
+                self.reconnect(false).await;
             },
         }
 
         Ok(())
     }
 
-    async fn reconnect(&mut self) {
+    async fn reconnect(&mut self, resuming: bool) {
```

Comment:
I know this is for resume, however I'm not sure this entirely makes sense, as it leads to a lot of ``reconnect(false)`` when you want to reconnect, which at a glance, looks like you don't want to reconnect like [here](https://github.com/dawn-rs/dawn/pull/94/files#diff-88211fc5e654c03fd0be06a8f305a9aeR241). 

Maybe this makes more sense to be reverse? So when you'd want to reconnect you'd do ``reconnect(true)``?

Just curious on thoughts here, it's an internal function so it's not a huge deal.

Reply:
Yeah it may make more sense to reverse it, the other solution would be to have a enum like:
```rust
pub enum ReconnectionKind {
    Reconnect,
    Resume,
}
```
but that would lead to code like this:
```rust
self.reconnect(ReconnectionKind::Reconnect).await;
```
so just reversing it may make more sense, what is your thoughts on this?

Reply:
hmm I think an enum makes more sense, as it's really clear what it's doing, however i think reversing it should be enough in this case.